File: c++/leetcode/1351. Count Negative Numbers in a Sorted Matrix.cpp
class Solution
{
public:
  int countNegatives(vector<vector<int>> &grid)
  {
    int found = 0;

    for (auto i : grid)
    {
      int pos = i.size() - 1;
      for (int j = pos; j >= 0; j--)
      {
        if (i[j] >= 0)
          break;
        found++;
      }
    }

    return found;
  }
};

// could use binary search to optimize inner loop

File: javascript/leetcode/1351. Count Negative Numbers in a Sorted Matrix.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var countNegatives = function (grid) {
  let found = 0;

  for (let i of grid) {
    const pos = i.length - 1;
    for (let j = pos; j >= 0; j--) {
      if (i[j] >= 0) break;
      found++;
    }
  }

  return found;
};

// could use binary search to optimize inner loop
